x/callback/abci.go

```diff
@@ -0,0 +1,94 @@
+package callback
+
+import (
+	"fmt"
+	"runtime/debug"
+
+	"cosmossdk.io/collections"
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/libs/log"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/archway-network/archway/x/callback/keeper"
+	"github.com/archway-network/archway/x/callback/types"
+)
+
+// EndBlocker fetches all the callbacks registered for the current block height and executes them
+func EndBlocker(ctx sdk.Context, k keeper.Keeper, wk types.WasmKeeperExpected) []abci.ValidatorUpdate {
+	logger := k.Logger(ctx)
+	params, err := k.Params.Get(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	currentHeight := ctx.BlockHeight()
+	// fetching all callbacks for current height
+	callbacks, err := k.GetCallbacksByHeight(ctx, currentHeight)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, callback := range callbacks {
+		// creating CallbackMsg which is encoded to json and passed as input to contract execution
+		callbackMsg := types.NewCallbackMsg(callback.GetJobId())
+		// handling any panics
+		defer recoverAnyPanics(logger, callback)()
```

Comment:
deferring in a for loop will not work as expected;

we should execute a callback in a separate function and put the defer recover there